pkg/ipam/types.go

```diff
@@ -114,3 +115,19 @@ func (ipam *IPAM) DebugStatus() string {
 	ipam.allocatorMutex.RUnlock()
 	return str
 }
+
+// GetVpcCIDRs returns all the CIDRs associated with the VPC this node belongs to.
+// This works only cloud provider IPAM modes and returns nil for other modes.
+func (ipam *IPAM) GetVpcCIDRs() (vpcCIDRs []*cidr.CIDR) {
+	sharedNodeStore.mutex.RLock()
+	defer sharedNodeStore.mutex.RUnlock()
+	primary, secondary := deriveVpcCIDRs(sharedNodeStore.ownNode)
```

Comment:
Ah so by the time we get to this point here, we reference an already-populated `sharedNodeStore.ownNode` from the daemon `startIPAM()`, which is called before we even attempt to restore the cilium_host IPs. :+1: 

It would probably be good to document this assumption somewhere in the code.

Reply:
Added comments to  document the same.